src/semidiscretization/semidiscretization_coupled.jl

```diff
@@ -547,7 +547,8 @@ end
 
 function copy_to_coupled_boundary!(u_ode, semi_coupled, semi, i, n_boundaries,
                                    boundary_condition, boundary_conditions...)
-    copy_to_coupled_boundary!(boundary_condition, u_ode, semi_coupled, semi)
+    other_semi = semi_coupled.semis[boundary_condition.other_semi_index]
```

Comment:
By defining `other_semi`, we get a type instability here. But then the performance-relevant code further down is type-stable.